superdesk/core/elastic/resources.py

```diff
@@ -0,0 +1,398 @@
+        :raises ValueError: If no registered resources found
+        :raises ValueError: If the search is destined for different Elasticsearch clusters
+        :raises KeyError: If any provided resource name is not registered for use with Elasticsearch
+        """
+
+        resource_configs = set(
+            [
+                config.elastic.prefix
+                for config in self.app.resources.get_all_configs()
+                if config.elastic is not None and config.name in resource_names
+            ]
+        )
+
+        if len(resource_configs) == 0:
+            raise ValueError("Resources not found")
+        if len(resource_names) > 1:
+            raise ValueError("Multiple prefixes found, searching multiple clusters not supported")
+
+        client = self.get_client(resource_names[0])
+        indexes = [self.get_client(resource_name).config.index for resource_name in resource_names]
+        return client.search(query, indexes)
+
+    async def search_async(self, resource_names: List[str], query: Dict[str, Any]) -> Dict[str, Any]:
+        """Search Elasticsearch across multiple indexes
+
+        :param resource_names: A list of names of the registered resources to search in
+        :param query: The search query to filter items by
+        :return: A dictionary containing the results of the search
+        :raises ValueError: If no registered resources found
+        :raises ValueError: If the search is destined for different Elasticsearch clusters
+        :raises KeyError: If any provided resource name is not registered for use with Elasticsearch
+        """
+
+        resource_configs = set(
+            [
+                config.elastic.prefix
+                for config in self.app.resources.get_all_configs()
+                if config.elastic is not None and config.name in resource_names
+            ]
+        )
+
+        if len(resource_configs) == 0:
+            raise ValueError("Resources not found")
+        if len(resource_names) > 1:
+            raise ValueError("Multiple prefixes found, searching multiple clusters not supported")
+
+        client = self.get_client_async(resource_names[0])
+        indexes = [self.get_client(resource_name).config.index for resource_name in resource_names]
+        return await client.search(query, indexes)
+
+    def find_by_id(self, item_id: str, resource_names: List[str]) -> Optional[Dict[str, Any]]:
+        """Find a document in multiple registered resources and Elasticsearch indexes
+
+        :param item_id: The id of the item to find
+        :param resource_names: A list of names of the registered resources to search in
+        :return: The document if found, else None
+        :raises KeyError: If any provided resource name is not registered for use with Elasticsearch
+        """
+
+        for resource_name in resource_names:
+            doc = self.get_client(resource_name).find_by_id(item_id)
```

Comment:
in general it can be searching multiple indexes at the same time, as long as they are using the same connection

Reply:
Looks like the underlying request uses the [docs-get](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-get.html) endpoint of Elasticsearch, which by the look from their API reference (both in python code and that website) the index param only allows 1.

I'm thinking because Elastic is not actually searching, but returning a document directly based on ID. This should be fast enough. We also have the `app.elastic.search` function that allows searching across indexes of the same connection, if required.